fastlane/fastlane
# Updated Fastlane Swift package structure

### Checklist


### Motivation and Context

This is part one of a larger effort to revamp Fastlane Swift. You can read more about it in this proposal.

### Description

This pull request introduces an updated Fastlane Swift package structure.

**Hiding Unnecessary Folders in the Fastlane Swift Package**

You can prevent Xcode from displaying unwanted folders by adding an empty Package.swift file to them. This improves the developer experience when working on the package by reducing package loading time (I struggled to even open it) and simplifying the structure.

| Before | After  |
|--------|--------|
|  <img width="333" alt="image" src="https://github.com/user-attachments/assets/d519bb99-b298-4554-bc60-a9ae56ee1dfa"> |  <img width="296" alt="image" src="https://github.com/user-attachments/assets/aa82de49-7abd-428d-b7e7-9ea84176a991"> |

**New Executable Swift Package**

When a user executes `fastlane init swift`, a brand new Swift package is created. This package provides a `Fastfile` for writing lanes and a clean entry point powered by the swift-argument library.

<img width="638" alt="image" src="https://github.com/user-attachments/assets/bba6cbbe-ce80-47cf-945e-720809871cf7">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/306feb28-7d3f-4c1c-ab33-d56f38abc283">

**Removed `FastlaneSwiftRunner` Xcode project**

The `FastlaneSwiftRunner` Xcode project has been removed. From now on, we exclusively support the Swift package setup.

**Misc**
* Updated Swift tools to version 5.9 in the `Package.swift` manifest file.
* Removed the creation of `Fastlane.swift` and `Appfile.swift` files during the `fastlane init swift` setup, as they are now automatically included in the new Swift package.
* Updated paths for generating Fastlane Swift APIs.

### Testing Steps

1. Point your fastlane installation to this branch:

```ruby 
// Gemfile 

source "https://rubygems.org"

gem "fastlane", git: "https://github.com/fastlane/fastlane.git", branch: "swift-new-package-structure"
```

2. Execute
```shell
$ fastlane init swift 
```

3. Open and compile the Swift package at `./fastlane/swift/Package.swift` 🎉 

Please note that running fastlane will result in runtime errors. It's known error that's happening in the current SPM setup too. It will be addressed with future pull requests.
